mpf/devices/segment_display/segment_display.py

```diff
@@ -124,51 +142,38 @@ def add_text(self, text: str, priority: int = 0, key: str = None) -> None:
         """
         self.add_text_entry(TextStackEntry(text, None, None, None, None, None, priority, key))
 
-    def set_flashing(self, flashing: FlashingType, flash_mask: str = ""):
-        """Enable/Disable flashing."""
-        self.flashing = flashing
-        self.flash_mask = flash_mask
-        self._update_display(self.text)
-
-    def set_color(self, colors: List[RGBColor]):
-        """Set display colors."""
-        if not isinstance(colors, list):
-            colors = [colors]
-        self.colors = colors
-        assert self.hw_display is not None
-        self.hw_display.set_color(colors)
-        if self.virtual_connector:
-            self.virtual_connector.set_color(self.name, colors)
-
-    def remove_text_by_key(self, key: Optional[str] = None):
+    def remove_text_by_key(self, key: str):
         """Remove entry from text stack."""
         if key in self._text_stack:
             del self._text_stack[key]
-            if not self._text_stack:
-                self.add_text_entry(self.empty_text_stack_entry)
-            else:
-                self._update_stack()
+            self._update_stack()
 
     # pylint: disable=too-many-arguments
     def _start_transition(self, transition: TransitionBase, current_text: str, new_text: str,
-                          current_colors: Optional[List[RGBColor]] = None, new_colors: Optional[List[RGBColor]] = None,
-                          update_hz: float = 30.0):
+                          current_colors: List[RGBColor], new_colors: List[RGBColor],
+                          update_hz: float):
         """Start the specified transition."""
+        current_colors = self._expand_colors(current_colors, len(current_text))
```

Comment:
Same potential problem with the text length here (integrated commas/dots)

Reply:
Yeah we probably have to interleave color + text in the beginning. Or do we? I guess colors are just for the character positions and interleaved dots just get the same color. I will have a look tomorrow